devito/ir/iet/nodes.py

```diff
@@ -373,25 +388,24 @@ def dimensions(self):
     @property
     def is_scalar(self):
         """True if a scalar expression, False otherwise."""
-        return self.expr.lhs.is_Symbol
+        return isinstance(self.expr.lhs, (AbstractSymbol, IndexedBase))
 
     @property
     def is_tensor(self):
         """True if a tensor expression, False otherwise."""
-        return not self.is_scalar
+        return self.expr.lhs.is_Indexed
```

Comment:
Can't it be a tensor expression if lhs is tensor but rhs is not? Like something like computing a norm or something?

Reply:
docstring should be reformulated (will do now) -- what I mean here is whether we're writing to an entry in an array, which would implicitly mean that the Expression cannot be an initializing expression,